.github/workflows/release.yml

```diff
@@ -39,59 +32,47 @@ on:
         default: false
         type: boolean
 
-env:
-  TAG_NAME: v${{ inputs.version }}
-
 permissions:
   contents: read
 
 concurrency:
   group: ${{ github.workflow }}
 
+env:
+  RELEASE_VERSION: ${{ inputs.version }}
+
 jobs:
-  prepare_release:
-    permissions:
-      contents: write
-    name: "Changelog and Version Bump"
-    if: ${{ ! inputs.skip_preparation }}
+  validate-tag:
     runs-on: ubuntu-latest
     steps:
-      - uses: elastic/apm-pipeline-library/.github/actions/github-token@current
-        with:
-          url: ${{ secrets.VAULT_ADDR }}
-          roleId: ${{ secrets.VAULT_ROLE_ID }}
-          secretId: ${{ secrets.VAULT_SECRET_ID }}
-      - uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+      - name: Checkout
+        uses: actions/checkout@v4
         with:
-          username: ${{ env.GIT_USER }}
-          email: ${{ env.GIT_EMAIL }}
-          token: ${{ env.GITHUB_TOKEN }}
-      - uses: actions/checkout@v4
+          ref: ${{ inputs.ref }}
+          fetch-depth: 0
+      - name: Validate tag does not exist on current commit
+        uses: ./.github/workflows/validate-tag
         with:
-          ref: ${{ inputs.branch }}
-          token: ${{ env.GITHUB_TOKEN }}
-      - name: Install JDK and mvn clean
-        uses: ./.github/workflows/maven-goal
-        with:
-          command: ./mvnw clean
-      - name: Prepare changelog for release
-        if: ${{ inputs.update_changelog && ! inputs.dry_run }}
+          tag: v${{ env.RELEASE_VERSION }}
+      - name: Validate tag match current version
         run: |
-          java .ci/ReleaseChangelog.java CHANGELOG.asciidoc ${{ inputs.version }}
-          git commit -m "Prepare changelog for release ${{ inputs.version }}" CHANGELOG.asciidoc
-      - name: Bump version and add git tag
-        if: ${{ ! inputs.dry_run }}
-        run: ./mvnw release:prepare -B -DpushChanges=false "-Darguments=-DskipTests -Dmaven.javadoc.skip=true" -DreleaseVersion=${{ inputs.version }}
-      - name: Push changes
-        if: ${{ ! inputs.dry_run }}
-        run: git push --atomic origin ${{ inputs.branch }} ${{ env.TAG_NAME }}
+          if [ "$(mvn -q help:evaluate -Dexpression=project.version -DforceStdout)" != "${{ env.RELEASE_VERSION }}" ]; then
+            echo "Tag should match pom.xml project.version"
+            exit 1
+          fi
+      - name: Validate version is a release version
+        run: |
+          if [[ "$(./mvnw -q help:evaluate -Dexpression=project.version -DforceStdout)" =~ "-SNAPSHOT" ]]; then
+            echo "This is a snapshot version"
+            exit 1
+          fi
 
-  maven_central_deploy:
+  maven-central-deploy:
```

Comment:
**nit**:

What's the reason fo changing the name? I'm certainly biased with `underscores` for clarity.

Reply:
It's more about trying to align on cases and a preference rather than a real thing.